src/utils/wallet.js

```diff
@@ -211,6 +214,34 @@ const wallet = {
     }
   },
 
+  /**
+   * The wallet needs each token metadata to show information correctly
+   * So we fetch the tokens metadata and store on redux
+   *
+   * @param {Array} tokens Array of token uids
+   * @param {String} network Network name
+   *
+   * @memberof Wallet
+   * @inner
+   **/
+  async fetchTokensMetadata(tokens, network) {
+    const metadatas = {};
+
+    for (const token of tokens) {
```

Comment:
Do we have any reason for downloading metadata in series? 

Maybe we could parallelize this using Promise.all

Reply:
As we discussed in private, it's better to keep the serial download for now.